### PR TITLE
moved encrypt decrypt to common.go

### PR DESCRIPTION
**Pull request by davisgu**
This change is [Reviewable](https://reviewable.io/reviews/matthewstevenson88/grpc-go/24)

### CODE REVIEW

File: security/s2a/internal/crypter/common.go [hunk 1]
```diff
@@ -23,3 +28,40 @@ func sliceForAppend(in []byte, n int) (head, tail []byte) {
...
+
+	// Seal appends the ciphertext and the tag to its first argument and
+	// returns the updated slice. However, sliceForAppend above ensures that
+	// dst has enough capacity to avoid a reallocation and copy due to the
+	// append.
+	dst = aead.Seal(dst[:dlen], nonce, data, aad)
+	return dst, nil
+}
+
+func decrypt(aead cipher.AEAD, dst, ciphertext, nonce, aad []byte) ([]byte, error) {
```
Comment (matthewstevenson88) [inline security/s2a/internal/crypter/common.go:57]: I think we're missing a comment describing what `decrypt` does, similar to what exists for `encrypt`.

**Comment by matthewstevenson88** [comment]
Thanks for the PR Davis! Left one comment.

File: security/s2a/internal/crypter/common.go (hunk 1 above)
Comment (davisgu) [reply security/s2a/internal/crypter/common.go:59]: Done

Comment (matthewstevenson88) [reply security/s2a/internal/crypter/common.go:59]: Please add comments that describe the other arguments as well (dst, ciphertext, nonce, aad) similar to what exists for `Encrypt`.

Also, the [ALTS version does not have comments on `Decrypt`](https://github.com/grpc/grpc-go/blob/master/credentials/alts/internal/conn/aes128gcm.go), so once we agree on the wording here you can send a PR to @cesarghali that adds the comment there as well.

File: security/s2a/internal/crypter/common.go [hunk 2]
```diff
@@ -23,3 +28,44 @@ func sliceForAppend(in []byte, n int) (head, tail []byte) {
...
+	// returns the updated slice. However, sliceForAppend above ensures that
+	// dst has enough capacity to avoid a reallocation and copy due to the
+	// append.
+	dst = aead.Seal(dst[:dlen], nonce, data, aad)
+	return dst, nil
+}
+
+// decrypt is the decryption function for an AEAD crypter, where aead determines
+// the type of AEAD crypter, and dst the destination bytes for the decrypted ciphertext.
+// The dst, ciphertext, nonce, and aad will be passed into aead to be opened decrypted
```
Comment (matthewstevenson88) [inline security/s2a/internal/crypter/common.go:59]: Please change this last sentence to: `The dst buffer may fully overlap with plaintext or not at all.`

**Comment by davisgu** [comment]
@cesarghali

**Pull request merged**